mesonbuild/compilers/mixins/gnu.py

```diff
@@ -314,6 +314,9 @@ def use_linker_args(cls, linker: str, version: str) -> T.List[str]:
                 f'Unsupported linker, only bfd, gold, and lld are supported, not {linker}.')
         return [f'-fuse-ld={linker}']
 
+    def get_gprof_args(self) -> T.List[str]:
+        return ['-pg', '-no-pie']
```

Comment:
I'm uncomfortable putting -no-pie here. The other cases we have where options don't interact correctly (undef + asan with clang) we simply warn "this might now work correctly".

Reply:
Ah, ok, so besides removing 'no-pie' I'll need to figure out how to add that text...

Reply:
> I'm uncomfortable putting -no-pie here. The other cases we have where options don't interact correctly (undef + asan with clang) we simply warn "this might now work correctly".

Is that supposed to go in the docs somewhere or output by meson at some point

Reply:
Why is -pg incompatible with pie in the first place?